### PR TITLE
feat: Link `list.pairwise`, `list.of_fn`

### PR DESCRIPTION
**Pull request by YaelDillies**
---

Match https://github.com/leanprover-community/mathlib/pull/15026

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by kbuzzard** [comment]
I don't understand why this PR doesn't have `pairwise_of_fn` in it.

**Comment by YaelDillies** [comment]
Sorry, it just wasn't updated (and not marked `awaiting-review`).

File: Mathlib/Data/List/OfFn.lean [hunk 1]
```diff
@@ -242,6 +242,11 @@ theorem ofFn_fin_repeat {m} (a : Fin m → α) (n : ℕ) :
     Nat.add_mul_mod_self_right, Nat.mod_eq_of_lt (Fin.is_lt _)]
 #align list.of_fn_fin_repeat List.ofFn_fin_repeat
 
+@[simp] lemma pairwise_ofFn {R : α → α → Prop} {n} {f : Fin n → α} :
+  (ofFn f).Pairwise R ↔ ∀ ⦃i j⦄, i < j → R (f i) (f j) := by
+  simp only [pairwise_iff_nthLe, Fin.forall_iff, length_ofFn, nthLe_ofFn', Fin.mk_lt_mk]
+  exact ⟨λ h i hi j hj hij ↦ h _ _ hj hij, λ h i j hj hij ↦ h _ (hij.trans hj) _ hj hij⟩
```
Comment (eric-wieser) [inline Mathlib/Data/List/OfFn.lean:248]: mathport says
```suggestion
@[simp]
theorem pairwise_ofFn {R : α → α → Prop} {n} {f : Fin n → α} :
    (ofFn f).Pairwise R ↔ ∀ ⦃i j⦄, i < j → R (f i) (f j) := by
  simp only [pairwise_iff_nth_le, Fin.forall_iff, length_of_fn, nth_le_of_fn', Fin.mk_lt_mk]
  exact ⟨fun h i hi j hj hij => h _ _ hj hij, fun h i j hj hij => h _ (hij.trans hj) _ hj hij⟩
```

**Review by eric-wieser — APPROVED** [review]
bors d+

I think it's better not to make stylistic changes on top of the mathport output beyond the ones made in `start_port.sh`

**Pull request closed without merging**